forum/qa-theme/SnowFlat/js/selectLanguage.js

```diff
@@ -0,0 +1,72 @@
+document.addEventListener("DOMContentLoaded", function(event) {
+    if (typeof CKEDITOR != 'undefined') {
+        CKEDITOR.on( 'instanceReady', function() {
+            const categories = [
+                {category : 'HTML i CSS', language: 'xml'},
+                {category : 'C i C++', language: 'cpp'},
+                {category : 'JavaScript, jQuery, AJAX', language: 'jscript'},
+                {category : 'PHP, Symfony, Zend', language: 'php'},
+                {category : 'SQL, bazy danych', language: 'sql'},
+                {category : 'C# i .NET', language: 'csharp'},
+                {category : 'Java', language: 'java'},
+                {category : 'Python, Django', language: 'python'},
+            ]
+            const codeBlock = document.getElementById( 'cke_43' );
+		
+            if ( location.href.includes( '/ask' ) ) {

```

Comment:
Łapanie clicka na elemencie **codeBlock** ustawiłbym raz - pod koniec skryptu. W IFach sprawdzających podstronę tworzyłbym funkcję, którą przypisywałbym do **let** zadeklarowanego nad IFami. Dzięki temu, podpięcie będzie jednokrotne i trafi tam przygotowana funkcja z kodem zależnym od podstrony.

Reply:
W taki sposób?

`...
let listener = newListener(...(code)...);
...
function newListener(code)
{
codeBlock.addEventListener('click', () => {code});
}
`

Reply:
Po pierwsze, nad IFami utwórz zmienną, np. **let onClick;**, albo **let clickHandler;**; Po drugie, linijki z:
`codeBlock.addEventListener( 'click', () => {`
 zamień na **clickHandler = () => {**
Po trzecie, pod IFami zapisz:
`codeBlock.addEventListener( 'click', onClick );` , albo **clickHandler** (zależy, co utworzyłeś).

Chodzi po prostu o to, aby listener podpinać tylko raz, natomiast kod funkcji obsługujący zdarzenie utworzyć sobie wedle potrzeb w zależności od IFów.

Reply:
Spoko, rozumiem.